pkg/istio/rbac/interface.go

```diff
@@ -18,3 +18,9 @@ type Provider interface {
 	// GetCurrentIstioRbac returns the Istio RBAC custom resources associated with the given model
 	GetCurrentIstioRbac(model athenz.Model, csc model.ConfigStoreCache) []model.Config
 }
+
+type ProviderV2 interface {
+	// ConvertAthenzModelIntoIstioAuthzPolicy returns the Istio Authorization Policy Resource associated with given athenz model,
+	// namespace, service name and service label
+	ConvertAthenzModelIntoIstioAuthzPolicy(athenzModel athenz.Model, namespace string, serviceName string, svcLabel string) model.Config
```

Comment:
Any reason we cannot reuse the existing provider interface? I think we should implement ConvertAthenzModelIntoIstioRbac and GetCurrentIstioRbac functions so that we can compare the authz policies we generated with the existing ones on the cluster. This will allow us to vastly reduce the number of api calls we make. Similar to here: https://github.com/yahoo/k8s-athenz-istio-auth/blob/master/pkg/controller/controller.go#L175-L180

Reply:
added the function. but computeChangeList will be different, we cannot tell only from athenz domain how many authz policies created, maybe looping through authz files, regenerate and compare, and update those needed. Maybe we can add this logic in a later pr because we are enabling dry run only this time.